crates/chia-consensus/src/gen/solution_generator.rs

```diff
@@ -39,6 +40,46 @@ where
     Ok(quote)
 }
 
+// do not use generally
+fn len_for_generator(val: u64) -> usize {
+    if val < 0x80 {
+        0 // this encompasses a chialisp optimisation that happens for small ints
+    } else if val < 0x8000 {
+        2
+    } else if val < 0x0080_0000 {
+        3
+    } else if val < 0x8000_0000 {
+        4
+    } else if val < 0x0080_0000_0000 {
+        5
+    } else if val < 0x8000_0000_0000 {
+        6
+    } else if val < 0x0080_0000_0000_0000 {
+        7
+    } else if val < 0x8000_0000_0000_0000 {
+        8
+    } else {
+        9
+    }
+}
+
+// calculate the size in bytes of a generator with no backref optimisations
+pub fn calculate_generator_length<I>(spends: I) -> usize
+where
+    I: AsRef<[CoinSpend]>,
+{
+    let mut size: usize = 5; // (q . (())) => ff01ff8080 => 5 bytes
+
+    for s in spends.as_ref() {
+        let puzzle = s.puzzle_reveal.as_ref();
+        let solution = s.solution.as_ref();
+        // parent-id puzzle-reveal amount solution + bytes for list extension and atom prep bytes
+        size += 32 + puzzle.len() + len_for_generator(s.coin.amount) + solution.len() + 8;
```

Comment:
the 32 is the parent coin ID, right? the serialized length is 33 bytes (because of the length prefix). I suspect that the length prefix (as well as the missing byte from `len_for_generator()` is lumped into the `8` at the end. I think it would be easier to understand if they were more explicitly acknowledged in the code.